src/software/thunderscope/thunderscope_config.py

```diff
@@ -0,0 +1,563 @@
+            )
+        )
+
+    return TScopeConfig(proto_unix_io_map=proto_unix_io_map, tabs=tabs)
+
+
+def configure_ai_or_diagnostics(
+    load_blue,
+    load_yellow,
+    load_diagnostics,
+    visualization_buffer_size=5,
+    cost_visualization=False,
+):
+    """
+    Constructs a view with one of:
+        - 1 Fullsystem (Blue or Yellow)
+        - 1 Fullsystem (Blue or Yellow) and Diagnostics
+        - Diagnostics
+
+    :param load_blue: if blue fullsystem should be loaded
+    :param load_yellow: if yellow fullsystem should be loaded
+    :param load_diagnostics: if diagnostics should be loaded
+    :param visualization_buffer_size: The size of the visualization buffer.
+            Increasing this will increase smoothness but will be less realtime.
+    :param cost_visualization: True if cost visualization widget should be enabled
+                                False if not
+    :return: the Thunderscope Config for this view
+    """
+
+    def get_extra_widgets(proto_unix_io):
+        """
+        Gets the extra widgets for the fullsystem tab
+        :param proto_unix_io: the proto unix io to configure widgets with
+        :return: list of widget data for the extra widgets
+        """
+        extra_widgets = (
+            [configure_cost_vis(proto_unix_io)] if cost_visualization else []
+        )
+        extra_widgets.append(configure_robot_view_fullsystem(proto_unix_io))
+        return extra_widgets
+
+    proto_unix_io_map = {ProtoUnixIOTypes.SIM: ProtoUnixIO()}
+    tabs = []
+
+    # Must be called before widgets are initialized below
+    initialize_application()
+
+    if load_blue:
+        proto_unix_io_map[ProtoUnixIOTypes.BLUE] = ProtoUnixIO()
+        proto_unix_io_map[ProtoUnixIOTypes.CURRENT] = proto_unix_io_map[
+            ProtoUnixIOTypes.BLUE
+        ]
+        tabs.append(
+            TScopeQTTab(
+                name="Blue Fullsystem",
+                key=TabNames.BLUE,
+                widgets=configure_base_fullsystem(
+                    full_system_proto_unix_io=proto_unix_io_map[ProtoUnixIOTypes.BLUE],
+                    sim_proto_unix_io=proto_unix_io_map[ProtoUnixIOTypes.SIM],
+                    friendly_colour_yellow=False,
+                    visualization_buffer_size=visualization_buffer_size,
+                    extra_widgets=get_extra_widgets(
+                        proto_unix_io_map[ProtoUnixIOTypes.BLUE]
+                    ),
+                ),
+            )
+        )
+    elif load_yellow:
+        proto_unix_io_map[ProtoUnixIOTypes.YELLOW] = ProtoUnixIO()
+        proto_unix_io_map[ProtoUnixIOTypes.CURRENT] = ProtoUnixIOTypes.YELLOW
+        tabs.append(
+            TScopeQTTab(
+                name="Yellow Fullsystem",
+                key=TabNames.YELLOW,
+                widgets=configure_base_fullsystem(
+                    full_system_proto_unix_io=[ProtoUnixIOTypes.YELLOW],
+                    sim_proto_unix_io=proto_unix_io_map[ProtoUnixIOTypes.SIM],
```

Comment:
There seems to be a bug here that crashes when I try to launch with `--run_yellow`:
```suggestion
        proto_unix_io_map[ProtoUnixIOTypes.YELLOW] = ProtoUnixIO()
        proto_unix_io_map[ProtoUnixIOTypes.CURRENT] = proto_unix_io_map[ProtoUnixIOTypes.YELLOW]
        tabs.append(
            TScopeQTTab(
                name="Yellow Fullsystem",
                key=TabNames.YELLOW,
                widgets=configure_base_fullsystem(
                    full_system_proto_unix_io=proto_unix_io_map[ProtoUnixIOTypes.YELLOW],
                    sim_proto_unix_io=proto_unix_io_map[ProtoUnixIOTypes.SIM],
```